src/test/java/com/klarna/hiverunner/ViewTest.java

```diff
@@ -0,0 +1,132 @@
+            .append("id int, ")
+            .append("value string")
+            .append(")")
+            .toString());
+
+    shell
+        .execute(new StringBuilder()
+            .append("create table test_db.tableB (")
+            .append("id int, ")
+            .append("value string")
+            .append(")")
+            .toString());
+
+    shell.insertInto("test_db", "tableA").addRow(1, "v1").addRow(2, "v2").commit();
+    shell.insertInto("test_db", "tableB").addRow(1, "v3").addRow(2, "v4").commit();
+    
+    // Using alias names
+    shell
+        .execute(new StringBuilder()
+            .append("create view test_db.test_view1 ")
+            .append("as select 1 from test_db.tableA a ")
+            .append("join test_db.tableB b ")
+            .append("on a.id = b.id;")
+            .toString());
+
+    // Using all lowercase
+    shell
+        .execute(new StringBuilder()
+            .append("create view test_db.test_view2 ")
+            .append("as select 1 from test_db.tablea ")
+            .append("join test_db.tableb ")
+            .append("on tablea.id = tableb.id;")
+            .toString());
+
+    shell.executeStatement("select * from test_db.test_view1");
+    shell.executeStatement("select * from test_db.test_view2");
+  }
+
+  @Test
+  public void createViewMixedCases() {
+    
+    //creating tables
+    shell.execute("create database test_db");
+
+    shell
+        .execute(new StringBuilder()
+            .append("create table test_db.tableA (")
+            .append("id int, ")
+            .append("value string")
+            .append(")")
+            .toString());
+
+    shell
+        .execute(new StringBuilder()
+            .append("create table test_db.tableB (")
+            .append("id int, ")
+            .append("value string")
+            .append(")")
+            .toString());
+
+    shell.insertInto("test_db", "tableA").addRow(1, "v1").addRow(2, "v2").commit();
+    shell.insertInto("test_db", "tableB").addRow(1, "v3").addRow(2, "v4").commit();
+
+    // Using mixed case in create VIEW statement (with a JOIN ON construction)
+    shell
+        .execute(new StringBuilder()
+            .append("create view test_db.test_view3 ")
+            .append("as select 1 from test_db.tableA ")
+            .append("join test_db.tableB ")
+            .append("on tableA.id = tableB.id;")
+            .toString());
+
+    // Even more mixed cases
+    shell
+        .execute(new StringBuilder()
+            .append("CREAte view test_db.test_View4 ")
+            .append("as select 1 from test_db.tABleA ")
+            .append("join test_db.TABLEB ")
+            .append("on tAbleA.id = tabLeB.id;")
+            .toString());
+    
+    // Even MORE mixed cases
+    shell
+        .execute(new StringBuilder()
+            .append("CREAte viEW tEst_Db.test_ViEw5 ")
+            .append("as select 1 from tesT_db.tABleA ")
+            .append("join teSt_db.TABLEB ")
```

Comment:
Maybe mix the case of `join` and `on` too?

Reply:
Very good catch :) there was an error in the line `int index = hiveql.indexOf(" on ");` if `on` wasn't lower case